lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/transform/GeoProjection.java

```diff
@@ -60,7 +60,11 @@ public GeoProjection setGeoCalculator(GeoCalculator geoCalculator) {
 
     public final GeoCalculator getGeoCalculator() {
         if (geoCalculator == null) {
-            setGeoCalculator(new SimpleGeoCalculator());
+            synchronized (this) {
+                if (geoCalculator == null) {
+                    setGeoCalculator(new SimpleGeoCalculator());
+                }
+            }
         }
```

Comment:
Why not just 

```java
public final GeoCalculator getGeoCalculator() {
   synchronized (this) {
      if (geoCalculator == null) {
         setGeoCalculator(new SimpleGeoCalculator());
      }   
   }
   return geoCalculator;
}
```
?

Reply:
Because, after initial initialization, `geoCalculator` will always be non-null and then the synchronization overhead introduced by the synchronized block can be avoided. So it's a very minor performance optimization.